src/state/auth-jwt.ts

```diff
@@ -11,6 +17,45 @@ export const authJwtState = atom<AuthJwt | undefined>({
   default: undefined,
 });
 
+////////////////
+/// UserInfo ///
+////////////////
+
+export interface UserInfo {
+  username: string;
+  email: string;
+  phoneNumber?: string;
+}
+
+export const userInfoStateAsync = selector<UserInfo | undefined>({
+  key: 'userInfoStateAsync',
+  get: async ({ get }) => {
+    const authJwt = get(authJwtState);
+
+    if (!isAuthJwt(authJwt)) {
+      return undefined;
+    }
+
+    // we unfortuantely cannot use `useFetchWrapper` since that adds hooks (breaks rule of hooks)
+    // either we replace this selector with effects in a nested component that updates this atom
```

Comment:
i've seen people put a `RecoilEffectListener` component as a sibling of `App` and then that file basically has a component that has an effect that sets up the dep chain.

(e.g. if atomB's result depends on atomA)
```ts
const fetchWrapper = useFetchWrapper();

useEffect(() => {
   (async () => {
     const someResult = await fetchWrapper.get(...);
     setAtomB(someResult);
   })();
}, [atomA])
```

that way, we could use any hook we want bc it lives in a component.
I won't be doing this atm but if growing pains occur due to async selectors/atoms, we can move to this pattern

Reply:
That makes sense :thumbsup:

Reply:
That's actually pretty cool